c++/src/kj/async-inl.h

```diff
@@ -2073,6 +2104,9 @@ template <class T, class... Args>
 struct coroutine_traits<kj::Promise<T>, Args...> {
   // `Args...` are the coroutine's parameter types.
 
+  static_assert((!kj::_::isDisallowedInCoroutine<Args>::value && ...),
+      "Disallowed type in coroutine");
+
   static_assert((::kj::_::NoWaitScope<Args> && ...),
```

Comment:
Perhaps we should mark `WaitScope` with this annotation and then remove the separate `static_assert` for it?

Reply:
We certainly can. Let's make sure we're set on the basic implementation first.